### DISCUSSION TITLE
Translation of command

### DISCUSSION THREAD

**Discussion by jptrzy** [evennia/evennia#2583] [Evennia Questions] [answered]
How can I translate commands, for example, "inventory" to "inwentarz"?

**Comment by Griatch** [comment, marked as answer]
Commands are made to be replaced and customized. So you need to replace the default command with your own, translated version. Here's how you replace the default "inventory" with "inwentarz":

1. Copy and paste the default `CmdInventory` class code from `evennia/commands/default/general.py` to a place in your `mygame` folder (let's say in `mygame/commands/command.py`). You will have to also add some of the imports from that file that the command expects, such as the `COMMAND_DEFAULT_CLASS` (just copy the definition of this from the top of the `general.py` module). 
2. Now you want to make sure Evennia uses your copy. Since your new command will have a different key (`inwentarz` instead of `inventory`), you can't just add your version and have it replace the default - you must actively remove the old one. Make the following changes to `mygame/default_cmdsets`:

```python
from commands import command    # wherever you put your new command

# ... 
class CharacterCmdSet(default_cmds.CharacterCmdSet):
    # ...
    def at_cmdset_creation(self):
        # ...
        # first we remove the old default command ...
        self.remove(default_cmds.CmdInventory)
        # ... and add your copy instead! 
        self.add(command.CmdInventory())
```

3. Reload your server and make sure your inventory command still works (it is now using your copied version but you haven't changed it yet - this is just to make sure you copied everything correctly). 
4. Now you can edit your copied `CmdInventory` class, changing its `key` to `inwentarz` and translating any other strings it returns. 
5. Reload again and if all went well, your `inventory` command is now `inwentarz`.

**Reply by jptrzy** [reply]
Thank you for your answer, but could I do this using [i18n](https://www.evennia.com/docs/1.0-dev/Concepts/Internationalization.html).

django.po
```
#: commands/default/general.py:370
msgid "inventory"
msgstr "ekwipunek"

#: commands/default/general.py:379
msgid "You are not carrying anything."
msgstr "Nie masz przy sobie niczego."
```

I had tried adding those lines, and compile with `django-admin compilemessages`, reinstall, and reload server, but this doesn't work for me.

**Reply by Griatch** [reply]
No, i18n is only for hard-coded strings in the server, not for Commands.

**Comment by jptrzy** [comment]
I fond it! 

Text in **commands/default/general.py** wasn't properly set for translation, and that's why I couldn't see any results.
I could go and change **commands/default/general.py** to 
```
from django.utils.translation import gettext as _
.
.
.
key = _("inventory")
.
.
.
string = _("You are not carrying anything.")
```
and then it will work, but I think I will stay with the last suggestion.

**Reply by Griatch** [reply]
No, we don't translate commands, the lack of i18n in default commands is deliberate. The reason is that commands have their own replacement mechanism and you are likely to change them around as part of developing your game anyway. See https://www.evennia.com/docs/latest/Internationalization.html#translating-evennia for the policy.
